rclpy/rclpy/utilities.py

```diff
@@ -71,9 +72,37 @@ def get_rmw_implementation_identifier():
 
 
 def get_available_rmw_implementations():
-    """Return the set of all available RMW implementations as registered in the ament index."""
-    rmw_implementations = ament_index_python.get_resources('rmw_typesupport')
-    return {name for name in rmw_implementations if name != 'rmw_implementation'}
+    """
+    Return the set of all available RMW implementations as registered in the ament index.
+
+    The result can be overridden by setting an environment variable named
+    ``RMW_IMPLEMENTATIONS``.
+    The variable can contain RMW implementation names separated by the platform
+    specific path separator.
+    Including an unavailable RMW implementation results in a RuntimeError.
+    """
+    available_rmw_implementations = ament_index_python.get_resources(
+        'rmw_typesupport')
+    available_rmw_implementations = {
+        name for name in available_rmw_implementations
+        if name != 'rmw_implementation'}
+
+    # filter by implementations in environment variable if provided
+    rmw_implementations = os.environ.get('RMW_IMPLEMENTATIONS')
+    if rmw_implementations:
+        rmw_implementations = rmw_implementations.split(os.pathsep)
+        missing_rmw_implementations = set(rmw_implementations) - \
+            available_rmw_implementations
+        if missing_rmw_implementations:
+            raise RuntimeError(
+                f'The RMW implementations {missing_rmw_implementations} '
+                "specified in 'RMW_IMPLEMENTATIONS' are not available ("
+                ', '.join(sorted(available_rmw_implementations)) + ')')
```

Comment:
I don't think this code works as expected. The following seems to do the trick:
```suggestion
                f"{', '.join(sorted(available_rmw_implementations))})")
```

Reply:
I fixed it differently in 906fd4739e49c1cafe73d21e6f51d0860b63d8a1. With this I get an exception message like this:

> RuntimeError: The RMW implementations {'foo'} specified in 'RMW_IMPLEMENTATIONS' are not available (rmw_cyclonedds_cpp, rmw_fastrtps_cpp, rmw_fastrtps_dynamic_cpp)